src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -62,6 +62,17 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
             HtmlEncoder = htmlEncoder;
         }
 
+        public override async Task<bool> HandleRequestAsync()
+        {
+            if (Options.SignOutPath.HasValue && Options.SignOutPath == Request.Path)
```

Comment:
You should consider adding a tiny HTTP method constraint somewhere. GET is a must, but I guess POST could be also supported (that would be triggered from an auto-post form in an iframe).